src/main/java/com/Nearsoft/referrals/service/MailerService.java

```diff
@@ -1,5 +1,8 @@
 package com.Nearsoft.referrals.service;
 
+import javax.mail.MessagingException;
+import java.io.IOException;
+
 public interface MailerService {
-    void sendEmail(Long recruiterId, Long jobId, String referredName, String referredEmail);
+    void sendEmail(Long recruiterId, Long jobId, String referredName, String referredEmail, String fileName, StorageService storageService) throws MessagingException, IOException;
```

Comment:
Use parameters to pass data across methods not service instances... inject service instances using the spring mechanisms